tests_and_analysis/test/euphonic_test/test_force_constants_interpolate_phonons.py

```diff
@@ -0,0 +1,227 @@
+        image_data = np.loadtxt(
+            os.path.join(self.path, 'LZO', 'lzo_sc_image_i.txt')
+        )
+        i, j, n = get_ijn_from_image_data(image_data)
+        sc_i = image_data[:, 3].astype(int)
+        max_n = np.max(n) + 1
+        # size = n_atoms X n_atoms*n_cells_in_sc X max supercell images
+        expected_sc_image_i = np.full((22, 88, max_n), -1)
+        expected_sc_image_i[i, j, n] = sc_i
+        # After refactoring where the shape of sc_image_i was changed from
+        # (n_atoms, n_cells_in_sc*n_atoms, (2*lim + 1)**3) to
+        # (n_cells_in_sc, n_atoms, n_atoms, (2*lim + 1)**3),
+        # expctd_image_i must be reshaped to ensure tests still pass
+        expected_sc_image_i = np.transpose(
+            np.reshape(expected_sc_image_i, (22, 4, 22, max_n)),
+            axes=[1, 0, 2, 3]
+        )
+        filename = os.path.join(self.path, 'LZO', 'La2Zr2O7.castep_bin')
+        return ForceConstants.from_castep(filename), "_sc_image_i", \
+            expected_sc_image_i
+
+    @pytest.fixture
+    def graphite_calculate_supercell_images_sc_image_i(self):
+        image_data = np.loadtxt(
+            os.path.join(self.path, 'graphite', 'graphite_sc_image_i.txt')
+        )
+        i, j, n = get_ijn_from_image_data(image_data)
+        sc_i = image_data[:, 3].astype(int)
+        max_n = np.max(n) + 1
+        # size = n_atoms X n_atoms*n_cells_in_sc X max supercell images
+        expected_sc_image_i = np.full(
+            (self.n_atoms, self.n_atoms * self.n_cells_in_sc, max_n), -1)
+        expected_sc_image_i[i, j, n] = sc_i
+        # After refactoring where the shape of sc_image_i was changed from
+        # (n_atoms, n_cells_in_sc*n_atoms, (2*lim + 1)**3) to
+        # (n_cells_in_sc, n_atoms, n_atoms, (2*lim + 1)**3),
+        # expctc_image_i must be reshaped to ensure tests still pass
+        expected_sc_image_i = np.transpose(
+            np.reshape(
+                expected_sc_image_i,
+                (self.n_atoms, self.n_cells_in_sc,self.n_atoms, max_n)
+            ), axes=[1, 0, 2, 3]
+        )
+        filename = os.path.join(self.path, 'graphite', 'graphite.castep_bin')
+        return ForceConstants.from_castep(filename), "_sc_image_i", \
+            expected_sc_image_i
+
+    @pytest.mark.parametrize('calculate_supercell_images', [
+        pytest.lazy_fixture('lzo_calculate_supercell_images_n_sc_images'),
+        pytest.lazy_fixture('lzo_calculate_supercell_images_sc_image_i'),
+        pytest.lazy_fixture('graphite_calculate_supercell_images_n_sc_images'),
+        pytest.lazy_fixture('graphite_calculate_supercell_images_sc_image_i')
+    ])
+    def test_calculate_supercell_images(self, calculate_supercell_images):
+        fc, testing_attribute, expected_attribute_data = \
+            calculate_supercell_images
+        fc._calculate_supercell_images(self.lim)
+        npt.assert_equal(
+            getattr(fc, testing_attribute), expected_attribute_data
+        )
```

Comment:
I'm wondering if we should be testing these functions at all as they're all private, and it's more to maintain. Although they have already been written...

Reply:
Strictly we should only test public methods in unit testing, so I'd agree they should probably be removed